library/Freckle/App/Test/Hspec/Runner.hs

```diff
@@ -66,18 +65,21 @@ runWith config name spec = do
   evaluateSummary $ reliableSummary <> isolatedSummary
  where
   load = flip readConfig
-  junit filename changeConfig =
-    (spec `runJUnitSpec` ("/tmp/junit", filename)) . changeConfig
-  hspec _ changeConfig = runSpec spec . changeConfig
+  runner filename changeConfig =
+    (spec `runWithJUnit` ("/tmp/junit", filename)) . changeConfig
   noConcurrency x = x {configConcurrentJobs = Just 1}
 
-runJUnitSpec :: Spec -> (FilePath, String) -> Config -> IO Summary
-runJUnitSpec spec (path, name) config =
-  spec `runSpec` configWithJUnit junitConfig config
+runWithJUnit :: Spec -> (FilePath, String) -> Config -> IO Summary
+runWithJUnit spec (path, name) config = do
+  junitEnabled <- envJUnitEnabled
+  baseJUnitConfig <- envJUnitConfig
+  let modify = if junitEnabled then configWithJUnit $ withOverride baseJUnitConfig else id
```

Comment:
Are you doing this because you can't make `-f junit` work? I would've expected something like,

```hs
junitConfig <- withOverride <$> envJUnitConfig
spec `runSpec` configWithJunitAvailable junitConfig config
```

Reply:
Just saw you other comment. I'd really like to understand why this is not working, but I get it if you just want to go ahead this way anyway.